sequencer/src/state.rs

```diff
@@ -964,6 +965,56 @@ impl FromStr for FeeAmount {
     }
 }
 
+impl FromStringOrInteger for FeeAmount {
+    type Binary = U256;
+    type Integer = u64;
+
+    fn from_binary(b: Self::Binary) -> anyhow::Result<Self> {
+        Ok(Self(b))
+    }
+
+    fn from_integer(i: Self::Integer) -> anyhow::Result<Self> {
+        Ok(i.into())
+    }
+
+    fn from_string(s: String) -> anyhow::Result<Self> {
```

Comment:
Not sure if it's better there is a function for this in ethers. 

https://github.com/gakonst/ethers-rs/blob/51fe937f6515689b17a3a83b74a05984ad3a7f11/ethers-core/src/utils/mod.rs#L245

Reply:
I think one difference is that it allows to specify fractional values, but checks that there aren't any fractional wei. So could be nice to use it to get some errors handled and the extra functionality.

Reply:
[1b771e5](https://github.com/EspressoSystems/espresso-sequencer/pull/1443/commits/1b771e5f791d2b97481a2e45ff2929e79fee8f92)